src/gurt/telemetry.c

```diff
@@ -1075,6 +1075,61 @@ d_tm_compute_histogram(struct d_tm_node_t *node, uint64_t value)
 	return rc;
 }
 
+/**
+ * Set the given counter to the specified \a value
+ *
+ * \param[in]	metric	Pointer to the metric
+ * \param[in]	value	Sets the counter to this \a value
+ */
+void
+d_tm_set_counter(struct d_tm_node_t *metric, uint64_t value)
+{
+	if (unlikely(d_tm_shmem_root == NULL || metric == NULL))
+		return;
+
+	if (unlikely(metric->dtn_type != D_TM_COUNTER)) {
+		D_ERROR("Failed to set counter [%s] on item not a "
+			"counter.\n", metric->dtn_name);
+		return;
+	}
+
+	if (unlikely(metric->dtn_protect))
+		D_MUTEX_LOCK(&metric->dtn_lock);
+
+	metric->dtn_metric->dtm_data.value = value;
```

Comment:
IMO there should be a check here to verify that the new value is not <= the current value. Or at least not less than. I think it's important to enforce this in the producer in order to protect against misuse of counters which can lead to nonsensical metric values over time. Missed this in my last review.